src/organisms/Trips.js

```diff
@@ -0,0 +1,92 @@
+import React, { useEffect } from 'react';
+import Grid from '@material-ui/core/Grid';
+import BusCard from '../Molecules/BusCard.js';
+import InputFeild from '../Atoms/InputFeild.js';
+import RefreshIcon from '@material-ui/icons/Refresh';
+import Custombutton from '../Atoms/Custombutton.js';
+
+const Trips = (props) => {
+    const trips = props.tripdetails
+    const tempTrips = [...trips]
+    const [tripData, setTripData] = React.useState([]);
+    const [flag, setFlag] = React.useState('')
+    const [company, setCompany] = React.useState('')
+    const [temp] = React.useState(tempTrips);
+
+    useEffect(() => {
+        if (company) {
+            let searchdata = trips.filter((ele) => {
+                if (ele.busdetails.buscompany.companyName === company) {
+                    return true;
+                }
+                return false;
+            })
+            setTripData(searchdata);
+        }
+        else if (flag) {
+            setTripData(trips);
+        }
+        else {
+            setTripData(temp);
+        }
+    }, [flag, company]);
+
+    const sortBy = () => {
+        if (flag === '' || flag === 'true') {
+            trips.sort((a, b) => (b.busdetails.buscompany.rating - a.busdetails.buscompany.rating));
+            setFlag("false")
+        }
+        else {
+            trips.sort((a, b) => (a.busdetails.buscompany.rating - b.busdetails.buscompany.rating));
+            setFlag("true")
+        }
+    }
+
+    var companies = trips.map((ele) => {
+        return ele.busdetails.buscompany.companyName;
+    })
+    companies = Array.from(new Set(companies));
```

Comment:
you can just declare a set and add those items in map or use peek operator and get rid of iterating and declaring twice